tools/clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -1791,11 +1792,36 @@ void CodeGenFunction::EmitStoreThroughExtVectorComponentLValue(RValue Src,
       }
       Builder.CreateStore(Vec, VecDstPtr);
     } else {
-      for (unsigned i = 0; i < VecTy->getVectorNumElements(); i++) {
-        if (llvm::Constant *Elt = Elts->getAggregateElement(i)) {
-          llvm::Value *EltGEP = Builder.CreateGEP(VecDstPtr, {Zero, Elt});
-          llvm::Value *SrcElt = Builder.CreateExtractElement(SrcVal, i);
-          Builder.CreateStore(SrcElt, EltGEP);
+      // If the vector pointer comes from a call inst, assume it's a subscript.
+      // Do a load and then a store to avoid generating multiple stores.
+      if (llvm::CallInst *CI = llvm::dyn_cast<llvm::CallInst>(VecDstPtr)) {
+        llvm::Value *vec = llvm::UndefValue::get(VecTy);
+        llvm::Value *old_vec = Builder.CreateLoad(VecDstPtr);
```

Comment:
Should we be reusing the load of the vector above? It looks like it does things like set alignment which would be good. Is it possible we are going to get two loads now?

Reply:
Fixed it.